src/Symfony/Component/Form/Extension/Core/Type/CollectionType.php

```diff
@@ -75,6 +75,25 @@ public function finishView(FormView $view, FormInterface $form, array $options)
         if ($form->getConfig()->hasAttribute('prototype') && $view->vars['prototype']->vars['multipart']) {
             $view->vars['multipart'] = true;
         }
+
+        $prefixOffset =  -1;
+        // check if the entry type also defines a block prefix
+        if (isset($options['entry_options']['block_prefix'])) {
```

Comment:
At this stage, the option is already included in the entry config option, for short we could use the `else` code only for both cases. wdyt?

Reply:
I thought so, my reasoning was that the collection may be empty and that can prevent method calls, because it has to account for the prototype too. Which lead me to update the tests and the code once again, because that was not enough :).

Reply:
Ok but if the collection is empty we don't iterate, so I've updated as you suggested.